worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/legacycompat/BedBlockCompatibilityHandler.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.extent.clipboard.io.legacycompat;
+
+import com.sk89q.jnbt.IntTag;
+import com.sk89q.jnbt.Tag;
+import com.sk89q.worldedit.registry.state.DirectionalProperty;
+import com.sk89q.worldedit.registry.state.EnumProperty;
+import com.sk89q.worldedit.registry.state.Property;
+import com.sk89q.worldedit.world.block.BlockState;
+import com.sk89q.worldedit.world.block.BlockStateHolder;
+import com.sk89q.worldedit.world.block.BlockType;
+import com.sk89q.worldedit.world.block.BlockTypes;
+
+import java.util.Map;
+
+public class BedBlockCompatibilityHandler implements NBTCompatibilityHandler {
+
+    private static final DirectionalProperty FacingProperty;
+    private static final EnumProperty PartProperty;
+
+    static {
+        DirectionalProperty tempFacing;
+        EnumProperty tempPart;
+        try {
+            tempFacing = (DirectionalProperty) (Property<?>) BlockTypes.RED_BED.getProperty("facing");
+            tempPart = (EnumProperty) (Property<?>) BlockTypes.RED_BED.getProperty("part");
+        } catch (NullPointerException | IllegalArgumentException | ClassCastException e) {
+            tempFacing = null;
+            tempPart = null;
+        }
+        FacingProperty = tempFacing;
+        PartProperty = tempPart;
+    }
+
+    @Override
+    public <B extends BlockStateHolder<B>> boolean isAffectedBlock(B block) {
+        return block.getBlockType() == BlockTypes.RED_BED;
+    }
+
+    @Override
+    public <B extends BlockStateHolder<B>> B updateNBT(B block, Map<String, Tag> values) {
+        Tag typeTag = values.get("color");
+        if (typeTag instanceof IntTag) {
+            String bedType = convertBedType(((IntTag) typeTag).getValue());
+            if (bedType != null) {
+                BlockType type = BlockTypes.get("minecraft:" + bedType);
+                if (type != null) {
+                    BlockState state = type.getDefaultState();
+
+                    Property facingProp = type.getProperty("facing");
```

Comment:
It is worth noting my ide flags these as `Raw use of parameterized class 'Property'`. Where in this case it should be `Property<Direction>`.
I matched the usage from https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/legacycompat/SkullBlockCompatibilityHandler.java#L64
Would it be preferred for me to  be explicit about the parameterized type?

Reply:
I think it's fine to leave as is. Personally, I would prefer fixing it, but that can be painful.